dataplane/forwarding/fwdport/ports/cpu.go

```diff
@@ -167,8 +179,53 @@ func (p *cpuPort) punt(v interface{}) {
 	fwdport.Increment(p, packet.Length(), fwdpb.CounterId_COUNTER_ID_TX_ERROR_PACKETS, fwdpb.CounterId_COUNTER_ID_TX_ERROR_OCTETS)
 }
 
+func (p *CPUPort) puntRemotePort(v any) {
+	packet, ok := v.(fwdpacket.Packet)
+	if !ok {
+		fwdport.Increment(p, 1, fwdpb.CounterId_COUNTER_ID_TX_ERROR_PACKETS, fwdpb.CounterId_COUNTER_ID_TX_ERROR_OCTETS)
+		return
+	}
+
+	p.ctx.RLock()
+	var ingressPID *fwdpb.PortId
+	if port, err := fwdport.InputPort(packet, p.ctx); err == nil {
+		ingressPID = fwdport.GetID(port)
+	}
+	egressPID := fwdport.GetID(p)
+	if port, err := fwdport.OutputPort(packet, p.ctx); err == nil {
+		egressPID = fwdport.GetID(port)
+	}
+	hostPort, err := packet.Field(fwdpacket.NewFieldIDFromNum(fwdpb.PacketFieldNum_PACKET_FIELD_NUM_HOST_PORT_ID, 0))
+	if err != nil {
+		fwdport.Increment(p, packet.Length(), fwdpb.CounterId_COUNTER_ID_TX_ERROR_PACKETS, fwdpb.CounterId_COUNTER_ID_TX_ERROR_OCTETS)
+		return
+	}
+
+	response := &fwdpb.PacketOut{
+		Packet: &fwdpb.Packet{
+			InputPort:  ingressPID,
+			OutputPort: egressPID,
+			HostPort:   binary.BigEndian.Uint64(hostPort),
+			Frame:      packet.Frame(),
+		},
+	}
+
+	ps := p.ctx.CPUPortSink()
+	p.ctx.RUnlock()
+	if ps != nil {
+		timer := deadlock.NewTimer(deadlock.Timeout, fmt.Sprintf("Punting packet from port %v", p))
+		err := ps(response)
+		timer.Stop()
+		if err == nil {
+			return
+		}
+		log.Errorf("ports: Unable to punt packet, request %+v, err %v.", response, err)
```

Comment:
Should it be treated as tx error like when ps == nil?

Reply:
yes good catch, fixed